### PR TITLE
fix(j-s): Case Files

### PR DESCRIPTION
**Pull request by gudjong**
# Case Files

[Framlengingar: Birta rannsóknargögn úr fyrra máli](https://app.asana.com/0/1199153462262248/1203966424106784/f)

## What

- Displays parent case files in extended cases. This was broken some months back.
- Fixes handling of investigation files. In many cases, appeal related files were mistakenly handled as investigation files.

## Why

- Verified bugs.

## Screenshots / Gifs

image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/web/src/routes/Court/InvestigationCase/Ruling/Ruling.tsx [hunk 1]
```diff
@@ -136,12 +138,12 @@ const Ruling = () => {
             <PoliceRequestAccordionItem workingCase={workingCase} />
             <AccordionItem
               id="caseFileList"
-              label={`Rannsóknargögn (${workingCase.caseFiles?.length ?? 0})`}
+              label={`Rannsóknargögn (${caseFiles.length})`}
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Court/InvestigationCase/Ruling/Ruling.tsx:141]: Consider changing this to be a Contentful string

File: apps/judicial-system/web/src/routes/Court/RestrictionCase/Ruling/Ruling.tsx [hunk 2]
```diff
@@ -230,12 +232,12 @@ export const Ruling: React.FC<React.PropsWithChildren<unknown>> = () => {
             <PoliceRequestAccordionItem workingCase={workingCase} />
             <AccordionItem
               id="caseFileList"
-              label={`Rannsóknargögn (${workingCase.caseFiles?.length ?? 0})`}
+              label={`Rannsóknargögn (${caseFiles.length})`}
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Court/RestrictionCase/Ruling/Ruling.tsx:235]: Consider changing this to be a Contentful string

File: apps/judicial-system/web/src/routes/Prosecutor/InvestigationCase/Overview/Overview.tsx [hunk 3]
```diff
@@ -306,16 +309,11 @@ export const Overview: React.FC<React.PropsWithChildren<unknown>> = () => {
             </AccordionItem>
             <AccordionItem
               id="id_6"
-              label={`Rannsóknargögn ${`(${
-                workingCase.caseFiles ? workingCase.caseFiles.length : 0
-              })`}`}
+              label={`Rannsóknargögn ${`(${caseFiles.length})`}`}
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Prosecutor/InvestigationCase/Overview/Overview.tsx:312]: Consider changing this to be a Contentful string

File: apps/judicial-system/web/src/routes/Prosecutor/RestrictionCase/Overview/Overview.tsx [hunk 4]
```diff
@@ -348,16 +351,11 @@ export const Overview: React.FC<React.PropsWithChildren<unknown>> = () => {
             </AccordionItem>
             <AccordionItem
               id="id_6"
-              label={`Rannsóknargögn ${`(${
-                workingCase.caseFiles ? workingCase.caseFiles.length : 0
-              })`}`}
+              label={`Rannsóknargögn ${`(${caseFiles.length})`}`}
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Prosecutor/RestrictionCase/Overview/Overview.tsx:354]: Consider changing this to be a Contentful string

**Review by unakb — APPROVED** [review]

**Pull request merged**